hera_sim/tests/test_noise.py

```diff
@@ -1,117 +1,103 @@
 
-        # check the shape
-        self.assertEqual(nos_jy.shape, (lsts.size, freqs.size))
 
-        # check that it's actually noiselike
-        # why is the tolerance set to what it is?
-        np.testing.assert_allclose(np.average(nos_jy, axis=0), 0, atol=0.7)
+@pytest.mark.parametrize("model", ["powerlaw", "HERA"])
+def test_resample_Tsky_shape(freqs, lsts, tsky_powerlaw, tsky_from_model, model):
+    tsky = tsky_powerlaw if model == "powerlaw" else tsky_from_model
+    assert tsky.shape == (lsts.size, freqs.size)
 
-        # figure out a better comment for this part
-        scaling = np.average(Tsky, axis=0) / Jy2T
-        dt = 10.7
 
-        # XXX this test works just fine, but it breaks when testing the
-        # entire repo... can't figure out why
-        np.testing.assert_allclose(
-            np.std(nos_jy, axis=0) / scaling * np.sqrt(B * dt), 1.0, atol=0.1
+@pytest.mark.parametrize("model", ["powerlaw", "HERA"])
+def test_resample_Tsky_time_behavior(tsky_powerlaw, tsky_from_model, model):
+    if model == "powerlaw":
+        assert all(
+            np.all(tsky_powerlaw[0] == tsky_powerlaw[i])
+            for i in range(1, tsky_powerlaw.shape[0])
         )
-
-        # now do it again but with the integration time set by lsts
-        np.random.seed(0)
-        nos_jy = noise.sky_noise_jy(
-            lsts, freqs, Tsky_mdl=Tsky_mdl, omega_p=omega_p, integration_time=None
-        )
-
-        dt = u.sday.to("s") / lsts.size
-        np.testing.assert_allclose(
-            np.std(nos_jy, axis=0) / scaling * np.sqrt(B * dt), 1.0, atol=0.1
+    else:
+        assert all(
+            np.all(tsky_from_model[0] != tsky_from_model[i])
+            for i in range(1, tsky_from_model.shape[0])
         )
 
-        # once more with a manually set channel width
-        np.random.seed(0)
-        B = 1e8
-        dt = 10.7
-        nos_jy = noise.sky_noise_jy(
-            lsts,
-            freqs,
-            Tsky_mdl=Tsky_mdl,
-            channel_width=B,
-            integration_time=dt,
-            omega_p=omega_p,
-        )
-        np.testing.assert_allclose(
-            np.std(nos_jy, axis=0) / scaling * np.sqrt(B * dt), 1.0, atol=0.1
-        )
-
-        # tsky = noise.resample_Tsky(freqs,lsts,noise.HERA_Tsky_mdl['xx'])
-        # nos_jy = noise.sky_noise_jy(tsky, freqs, lsts)
-        # import uvtools, pylab as plt
-        # uvtools.plot.waterfall(nos_jy, mode='real'); plt.show()
-
 
-if __name__ == "__main__":
-    unittest.main()
+@pytest.mark.parametrize("model", ["powerlaw", "HERA"])
+def test_resample_Tsky_freq_behavior(tsky_powerlaw, tsky_from_model, model):
+    tsky = tsky_powerlaw if model == "powerlaw" else tsky_from_model
+    assert all(np.all(tsky[:, 0] != tsky[:, i]) for i in range(1, tsky.shape[1]))
+
+
+@pytest.mark.parametrize("channel_width", [None, 1e6, 1e8])
+@pytest.mark.parametrize("integration_time", [None, 10.7, 33.7])
+@pytest.mark.parametrize("aspect", ["mean", "std"])
```

Comment:
Since choosing between "aspects" here essentially branches the whole function, I think its cleaner just to make it two tests. You can make the `noise_Jy` a fixture so that it doesn't repeat.

Reply:
I think that doing this will actually make the test code messier, because of these lines:
```
channel_width = channel_width or np.mean(np.diff(freqs)) * units.GHz.to("Hz")
integration_time = integration_time or units.day.to("s") / lsts.size
```
Based on my understanding of how to parametrize fixtures, there isn't an obviously neat way of determining the state of the `channel_width` and `integration_time` variables from the parametrized fixture. If you know of a way to do this, then I'd be happy to make the change with your help.